Intermittent test failures in GitHub CI?

I'm seeing test failures on GitHub CI that I'm not seeing locally. Take this job: https://github.com/wildfly/wildfly.org/actions/runs/15274701305/job/42958127410

```
2025-05-27 12:02:58,357 ERROR [io.qua.roq.fro.run.RoqRouteHandler] (executor-thread-1) Error occurred while rendering the template [news_p28.md]: java.lang.NullPointerException: Cannot invoke "io.quarkus.arc.ArcContainer.getActiveContext(java.lang.Class)" because "<local2>" is null [Error Occurred After Shutdown]
2025-05-27 12:02:58,358 ERROR [io.qua.roq.gen.run.RoqGenerator] (vert.x-eventloop-thread-0) Roq request failed /news/page/30/ [Error Occurred After Shutdown]: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:8081
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:336)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:339)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:784)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

It looks like...something is being shutdown early. The only way I can reproduce it locally -- and it may be different, but the result is the same -- is if I do this:

```
$ QUARKUS_ROQ_GENERATOR_BATCH=true mvn clean test
```

I ran Maven with `-X -e` and did not see any extra meaningful messages, fwiw.

> [@janwesterkamp](https://github.com/janwesterkamp) is this related to: [#532 (comment)](https://github.com/quarkiverse/quarkus-roq/issues/532#issuecomment-2953207290)

@ia3andy may be it's a blocked port too, but it looks like port 8081 might be blocked here, as `Connection refused: localhost/127.0.0.1:8081` on the logs above says.

There is an error before on that log [line 3004+](https://github.com/wildfly/wildfly.org/actions/runs/15274701305/job/42958127410#step:4:3004) saying that the test instatntiation fails (`TestInstantiationException: Failed to create test instance ):

```
...
2025-05-27 12:02:58,253 INFO  [io.qua.roq.gen.run.RoqGenerator] (vert.x-eventloop-thread-0) Roq generated file news/page/23/index.html
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 16.79 s <<< FAILURE! -- in org.wildfly.site.RoqSiteTest
Error:  org.wildfly.site.RoqSiteTest.testGen -- Time elapsed: 0.016 s <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
	at io.quarkus.test.junit.QuarkusTestExtension.initTestState(QuarkusTestExtension.java:851)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:818)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: io.smallrye.mutiny.TimeoutException
	at io.smallrye.mutiny.operators.uni.UniBlockingAwait.await(UniBlockingAwait.java:64)
	at io.smallrye.mutiny.groups.UniAwait.atMost(UniAwait.java:65)
	at io.quarkiverse.roq.generator.runtime.RoqGenerator.generateBlocking(RoqGenerator.java:115)
	at io.quarkiverse.roq.generator.runtime.RoqGenerator_ClientProxy.generateBlocking(Unknown Source)
	at io.quarkiverse.roq.testing.RoqGeneratorTestCallback$RoqAndRollManager.afterConstruct(RoqGeneratorTestCallback.java:55)
	at io.quarkiverse.roq.testing.RoqGeneratorTestCallback.afterConstruct(RoqGeneratorTestCallback.java:23)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at io.quarkus.test.junit.AbstractTestWithCallbacksExtension.invokeCallbacks(AbstractTestWithCallbacksExtension.java:162)
	at io.quarkus.test.junit.AbstractTestWithCallbacksExtension.invokeAfterConstructCallbacks(AbstractTestWithCallbacksExtension.java:64)
	at io.quarkus.test.junit.QuarkusTestExtension.initTestState(QuarkusTestExtension.java:848)
	... 4 more

2025-05-27 12:02:58,290 INFO  [io.qua.roq.gen.run.RoqGenerator] (vert.x-eventloop-thread-0) Roq generated file news/page/22/index.html
2025-05-27 12:02:58,338 ERROR [io.qua.roq.gen.run.RoqGenerator] (vert.x-eventloop-thread-0) Roq request failed /news/page/27/ [Error Occurred After Shutdown]: io.vertx.core.http.HttpClosedException: Connection was closed

2025-05-27 12:02:58,346 ERROR [io.qua.roq.gen.run.RoqGenerator] (vert.x-eventloop-thread-0) Roq request failed /news/page/28/ [Error Occurred After Shutdown]: io.vertx.core.http.HttpClosedException: Connection was closed

2025-05-27 12:02:58,346 ERROR [io.qua.roq.gen.run.RoqGenerator] (vert.x-eventloop-thread-0) Roq request failed /news/page/26/ [Error Occurred After Shutdown]: io.vertx.core.http.HttpClosedException: Connection was closed

2025-05-27 12:02:58,347 ERROR [io.qua.roq.gen.run.RoqGenerator] (vert.x-eventloop-thread-0) Roq request failed /news/page/25/ [Error Occurred After Shutdown]: io.vertx.core.http.HttpClosedException: Connection was closed

2025-05-27 12:02:58,347 ERROR [io.qua.roq.gen.run.RoqGenerator] (vert.x-eventloop-thread-0) Roq request failed /news/page/29/ [Error Occurred After Shutdown]: io.vertx.core.http.HttpClosedException: Connection was closed

2025-05-27 12:02:58,355 ERROR [io.qua.roq.gen.run.RoqGenerator] (vert.x-eventloop-thread-0) Roq request failed /news/page/3/ [Error Occurred After Shutdown]: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:8081
Caused by: java.net.ConnectException: Connection refused
...
```

@jasondlee can you check for port 8081 (default Quarkus test port) and 8082 (RoqAndRoll default static server port) are available for testing?

--- a/roq-generator/runtime/src/main/java/io/quarkiverse/roq/generator/runtime/RoqGenerator.java
+++ b/roq-generator/runtime/src/main/java/io/quarkiverse/roq/generator/runtime/RoqGenerator.java
@@ -112,7 +112,7 @@ public class RoqGenerator implements Handler<RoutingContext> {
     }
 
     public Path generateBlocking() {
-        return generate().await().atMost(Duration.ofSeconds(10));
+        return generate().await().atMost(Duration.ofSeconds(config.timeout()));
     }
 
     public Uni<Path> generate() {

